File: 입국심사.py
def solution(n,times):
    left = 1
    right = max(times)*n

    answer = right 

    while(left <= right):
        mid = (left+right//2)
        total = sum(mid// f for f in times)

        if(total >= n):
            answer = mid
            right = mid - 1

        else: 
            left = mid+1 


    return answer 

File: 입국심사.cpp
#include <iostream>
#include <vector>
#include <algorithm>


using namespace std;


long long solution(int n, vector<int> times){

    long long left = 1;
    long long right = *max_element(times.begin(),times.end()) * (long long)n;
    long long answer = right;

    while (left <= right){

        long long mid = (left+right) / 2;

        long long people= 0;


        for(int time: times){
            people = mid / time;

            if(people >= n){
                break;
            }

        }
        if(people >= n){
            answer = mid;
            right = mid - 1;  
        }
        else{
            left = mid + 1;
        }
    }

    return answer;

}